### ISSUE TITLE
std::out_of_range exception in caching_machinery_t

### ISSUE THREAD

**Issue by Tryneus** [rethinkdb/rethinkdb#5519]
While running `table_fuzzer.py --servers 5 --changefeeds --kill` against `next`, I encountered the following crash:

```
2016-03-14T14:46:52.566106091 20.309534s error: Error in ../src/errors.cc at line 161:
2016-03-14T14:46:52.566639698 20.310067s error: Uncaught exception of type "std::out_of_range"
  what(): map::at
2016-03-14T14:46:52.567553585 20.310981s error: Backtrace:
2016-03-14T14:46:53.627323902 21.370754s error: Mon Mar 14 14:46:52 2016

1 [0x16e373b]: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:108
2 [0x1ba5f40]: backtrace_t::backtrace_t() at backtrace.cc:210 (discriminator 1)
3 [0x1ba72fc]: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:331
4 [0x1ba5bc7]: format_backtrace(bool) at backtrace.cc:205
5 [0x1ba9e7e]: report_fatal_error(char const*, int, char const*, ...) at errors.cc:85
6 [0x1baa199]: rdb_terminate_handler() at errors.cc:161 (discriminator 1)
7 [0x7efd80ba6d46]: /opt/gcc-4.9.2/lib64/libstdc++.so.6(+0x63d46) [0x7efd80ba6d46] at 0x7efd80ba6d46 (/opt/gcc-4.9.2/lib64/libstdc++.so.6)
8 [0x7efd80ba6d91]: /opt/gcc-4.9.2/lib64/libstdc++.so.6(+0x63d91) [0x7efd80ba6d91] at 0x7efd80ba6d91 (/opt/gcc-4.9.2/lib64/libstdc++.so.6)
9 [0x7efd80ba6da6]: /opt/gcc-4.9.2/lib64/libstdc++.so.6(+0x63da6) [0x7efd80ba6da6] at 0x7efd80ba6da6 (/opt/gcc-4.9.2/lib64/libstdc++.so.6)
10 [0x7efd80ba6a02]: __cxa_call_unexpected+0x32 at 0x7efd80ba6a02 (/opt/gcc-4.9.2/lib64/libstdc++.so.6)
11 [0x1a3e2a3]: caching_cfeed_artificial_table_backend_t::caching_machinery_t::run(auto_drainer_t::lock_t) at caching_cfeed_backend.cc:124
12 [0x1a426ab]: void std::_Mem_fn<void (caching_cfeed_artificial_table_backend_t::caching_machinery_t::*)(auto_drainer_t::lock_t)>::operator()<auto_drainer_t::lock_t&, void>(caching_cfeed_artificial_table_backend_t::caching_machinery_t*, auto_drainer_t::lock_t&) const at functional:601
13 [0x1a42600]: void std::_Bind<std::_Mem_fn<void (caching_cfeed_artificial_table_backend_t::caching_machinery_t::*)(auto_drainer_t::lock_t)> (caching_cfeed_artificial_table_backend_t::caching_machinery_t*, auto_drainer_t::lock_t)>::__call<void, , 0ul, 1ul>(std::tuple<>&&, std::_Index_tuple<0ul, 1ul>) at functional:1295
14 [0x1a42586]: void std::_Bind<std::_Mem_fn<void (caching_cfeed_artificial_table_backend_t::caching_machinery_t::*)(auto_drainer_t::lock_t)> (caching_cfeed_artificial_table_backend_t::caching_machinery_t*, auto_drainer_t::lock_t)>::operator()<, void>() at functional:1353
15 [0x1a424cc]: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (caching_cfeed_artificial_table_backend_t::caching_machinery_t::*)(auto_drainer_t::lock_t)> (caching_cfeed_artificial_table_backend_t::caching_machinery_t*, auto_drainer_t::lock_t)> >::run_action() at callable_action.hpp:32
16 [0x1453296]: callable_action_wrapper_t::run() at runtime_utils.cc:48
17 [0x144c55d]: coro_t::run() at coroutines.cc:205 (discriminator 1)
18 [0x1a404fd]: coro_t* coro_t::spawn_sometime<std::_Bind<std::_Mem_fn<void (caching_cfeed_artificial_table_backend_t::caching_machinery_t::*)(auto_drainer_t::lock_t)> (caching_cfeed_artificial_table_backend_t::caching_machinery_t*, auto_drainer_t::lock_t)> >(std::_Bind<std::_Mem_fn<void (caching_cfeed_artificial_table_backend_t::caching_machinery_t::*)(auto_drainer_t::lock_t)> (caching_cfeed_artificial_table_backend_t::caching_machinery_t*, auto_drainer_t::lock_t)>&&) at coroutines.hpp:63
19 [0x1a3df7b]: caching_cfeed_artificial_table_backend_t::caching_machinery_t::caching_machinery_t(caching_cfeed_artificial_table_backend_t*) at caching_cfeed_backend.cc:62 (discriminator 2)
20 [0x1a3fe1f]: caching_cfeed_artificial_table_backend_t::construct_changefeed_machinery(signal_t*) at caching_cfeed_backend.cc:266
21 [0x1a4785a]: cfeed_artificial_table_backend_t::read_changes(ql::env_t*, ql::changefeed::streamspec_t const&, ql::backtrace_id_t, signal_t*, counted_t<ql::datum_stream_t>*, admin_err_t*) at cfeed_backend.cc:87
22 [0x1a43a0c]: artificial_table_t::read_changes(ql::env_t*, ql::changefeed::streamspec_t const&, ql::backtrace_id_t) at artificial_table.cc:88
23 [0x197a42f]: ql::changes_term_t::eval_impl(ql::scope_env_t*, ql::args_t*, ql::eval_flags_t) const at seq.cc:544
24 [0x18bc3cf]: ql::op_term_t::term_eval(ql::scope_env_t*, ql::eval_flags_t) const at op.cc:220
25 [0x177ea97]: ql::runtime_term_t::eval_on_current_stack(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:274
26 [0x177ef3c]: /home/ssd2/grey/rethinkdb/build/debug_clang/rethinkdb() [0x177ef3c] at 0x177ef3c ()
27 [0x177ecf1]: /home/ssd2/grey/rethinkdb/build/debug_clang/rethinkdb() [0x177ecf1] at 0x177ecf1 ()
28 [0x177ec7b]: ql::runtime_term_t::eval(ql::scope_env_t*, ql::eval_flags_t) const at term.cc:295
29 [0x19c86c5]: ql::query_cache_t::ref_t::run(ql::env_t*, ql::response_t*) at query_cache.cc:263 (discriminator 1)
```
...

**Comment by danielmewes** [comment]
Fix in review 3529

**Comment by danielmewes** [comment]
Fixed in next 33fe126
